src/common/snippets/src/lowered/linear_ir.cpp

```diff
@@ -24,41 +24,49 @@ LinearIR::LinearIR(const std::shared_ptr<ov::Model>& model, Config config)
     ExpressionPtr last_param = nullptr;
     for (const auto& n : get_ordered_ops(model)) {
         constExprIt insertion_pos = m_lowered_ops.end();
-        std::shared_ptr<Expression> expr;
-        std::vector<TensorDescriptorPtr> input_tds;
-        for (const auto& in : n->inputs()) {
-            const auto& out = in.get_source_output();
-            const auto& parent_out_tds = m_node2expression_map[out.get_node_shared_ptr()]->get_outputs();
-            input_tds.push_back(parent_out_tds[out.get_index()]);
+        const auto expr = create_expression(n, model);
+
+        // Scalar should be on the Linear IR beginning after Parameters to have valid expression order after Loop passes.
+        // After these passes we must call pass MoveScalarToConsumer() to have a correct accuracy.
+        // For more details, please see the pass description
+        if (const auto& scalar = as_type_ptr<op::Scalar>(n)) {
+            if (scalar_pos == m_lowered_ops.end()) {
+                OPENVINO_ASSERT(last_param, "Scalars must be executed after Parameters");
+                scalar_pos = std::find(m_lowered_ops.begin(), m_lowered_ops.end(), last_param);
+            }
+            insertion_pos = std::next(scalar_pos);
         }
-        if (const auto& par = as_type_ptr<opset1::Parameter>(n)) {
-            auto io_expr = std::make_shared<IOExpression>(par, model->get_parameter_index(par));
-            m_io_lowered_ops.push_back(io_expr);
-            expr = io_expr;
-            last_param = expr;
-        } else if (const auto& res = as_type_ptr<opset1::Result>(n)) {
-            auto io_expr = std::make_shared<IOExpression>(res, model->get_result_index(res), input_tds);
+
+        if (const auto io_expr = std::dynamic_pointer_cast<IOExpression>(expr)) {
+            register_expression(expr);
```

Comment:
Not directly relevant to this PR, but let's fix anyway.
We have 2 similar methods `register_regular_expression` and `register_expression`:
1) "regular expression" - has distinctive and completely unrelated meaning, let's not use it in this context.
2) It seems that we don't really need two methods that do exactly the same thing. Can we just add an extra argument `is_io_expression = false`, and explicitly set it to `true` whenever we need to skip the check.